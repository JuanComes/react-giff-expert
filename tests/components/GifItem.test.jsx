import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe('Pruebas en <GifItem />', () => {

    const title = "AOE2";
    const url = "https://age-of-empires.com/arbalest.jpg";


    test('Debe de hacer match con el Snapshot', () => {


        // Renderiza el componente
        const { container } = render( <GifGridItem title={ title } url={ url } />);

        // Compara con el snapshot
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        
        render( <GifGridItem title={ title } url={ url } />);
        
        // expect(screen.getByRole("img").src).toBe(url);
        // expect(screen.getByRole("img").alt).toBe(title);

        const {src, alt} = screen.getByRole("img");
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })
    

    test('debe de mostrar el tÃ­tulo en el componente', () => {
        
        render( <GifGridItem title={ title } url={ url } />);
        
    
        expect(screen.getByText(title)).toBeTruthy();
    })

});

