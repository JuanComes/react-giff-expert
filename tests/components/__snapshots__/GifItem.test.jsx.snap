// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="AOE2"
      src="https://age-of-empires.com/arbalest.jpg"
    />
    <p>
      AOE2
    </p>
  </div>
</div>
`;
